addons/l10n_ar/models/account_invoice.py

```diff
@@ -0,0 +1,473 @@
+        string="Computed AFIP Concept",
+        help="A concept is suggested regarding the type of the products on the"
+        " invoice but it is allowed to force a different type if required.",
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    l10n_ar_force_afip_concept = fields.Selection(
+        selection=afip_invoice_concepts,
+        string="AFIP Concept",
+        readonly=True,
+        help='AFIP requires to report the kind of products related to the'
+        ' invoices. The possible AFIP concepts are:\n'
+        ' * 1 - Producto / Exportación definitiva de bienes\n'
+        ' * 2 - Servicios\n'
+        ' * 3 - Productos y Servicios\n'
+        ' * 4 - Otros (exportación)\n',
+    )
+    l10n_ar_afip_service_start = fields.Date(
+        string='AFIP Service Start Date',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    l10n_ar_afip_service_end = fields.Date(
+        string='AFIP Service End Date',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+
+    @api.multi
+    @api.depends(
+        'invoice_line_ids',
+        'invoice_line_ids.product_id',
+        'invoice_line_ids.product_id.type',
+        'l10n_ar_force_afip_concept',
+    )
+    def _compute_l10n_ar_afip_concept(self):
+        for rec in self.filtered(
+                lambda x: x.company_id.country_id.code == 'AR' and
+                x.l10n_latam_use_documents):
+            afip_concept = False
+            if rec.l10n_ar_force_afip_concept:
+                afip_concept = rec.l10n_ar_force_afip_concept
+            else:
+                afip_concept = rec._get_concept()
+            rec.l10n_ar_afip_concept = afip_concept
+
+    @api.multi
+    def _inverse_l10n_ar_afip_concept(self):
+        for rec in self:
+            if rec._get_concept() == rec.l10n_ar_afip_concept:
+                rec.l10n_ar_force_afip_concept = False
+            else:
+                rec.l10n_ar_force_afip_concept = rec.l10n_ar_afip_concept
+
+    @api.multi
+    def _get_concept(self):
+        """ Method to get the concept of the invoice considering the type of
+        the products on the invoice
+        """
+        self.ensure_one()
+        invoice_lines = self.invoice_line_ids
+        product_types = set([
+            x.product_id.type for x in invoice_lines
+            if x.product_id])
+        consumible = set(['consu', 'product'])
+        service = set(['service'])
+        mixed = set(['consu', 'service', 'product'])
+        # default value "product"
+        afip_concept = '1'
+        if product_types.issubset(mixed):
+            afip_concept = '3'
+        if product_types.issubset(service):
+            afip_concept = '2'
+        if product_types.issubset(consumible):
+            afip_concept = '1'
+        # on expo invoice you can mix services and products
+        if self.l10n_latam_document_type_id.code in ['19', '20', '21']:
```

Comment:
```suggestion
        if self.l10n_latam_document_type_id.code in ['19', '20', '21'] and afip_concept == '3':
```

To avoid the next `if`